mergify_engine/worker.py

```diff
@@ -1130,6 +1110,14 @@ async def start(self) -> None:
         await redis_utils.load_scripts(self._redis_stream)
         await migrations.run(self._redis_cache)
 
+        self._dedicated_workers_syncer_task = asyncio.create_task(
```

Comment:
Can we avoid having a task just for this?
I'm not uber confident that this is not going to be the source of race conditions in the future.
I'd prefer to have a simpler scheme where we check the latest update timestamp in the code path that uses it and refresh it if it's too old before using it.

Reply:
The reason behind the dedicated task is,
We have multiple consumers of this data, especially each worker task (so around 60 per-process in production)
In the worse case, we lookup in this set every 0.42s per worker task.
If I use the method using a timestamp and check for expiration, the 60 tasks may update the data at the same time for no value (And I don't think we want to introduce a lock to avoid the update concurrency).

Another solution could be refresh it AFTER using it, so other workers will still use old data during the refresh but don't update the cache multiple times in parallel. This version has exactly the same kind of future source of race conditions of the dedicated task, as workers may use not yet refreshed data.

The code already protects some kind of race conditions around this set. If the subscription and the type of worker (dedicated/shared) don't match, we just update this set in Redis and retry later until everything are updated correctly.